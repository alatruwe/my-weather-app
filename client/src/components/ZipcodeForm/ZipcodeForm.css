.zipcodeForm {
  display: flex;
  flex-direction: column;
}

label {
  font-size: 1.5em;
}

input,
button {
  border-radius: 10px;
  font-size: 1.5em;
}

input {
  text-align: right;
}

/* button animation */
.btn {
  width: auto;
  background: transparent;
  border: none;
}
.btn:after {
  content: "";
  display: block;
  width: 0%;
  padding-top: 5px;
  border-bottom: 1px solid black;
  transition: 0.5s;
}
.btn:hover:after,
.btn:focus:after {
  width: 100%;
}
