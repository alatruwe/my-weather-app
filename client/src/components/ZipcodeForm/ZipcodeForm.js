import React, { Component } from "react";
import ValidationError from "../ValidationError";
import "./ZipcodeForm.css";
class ZipcodeForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      zipcode: 0,
      touched: false,
      error: null,
    };
  }

  updateZipcode = (number) => {
    this.setState({ zipcode: number, touched: true });
  };

  // form validation
  // input is not 5 char long
  validateZipcode() {
    const zipcode = this.state.zipcode;
    const zipcodeString = zipcode.toString();
    if (zipcodeString.length !== 5) {
      return "Please enter a valid zipcode";
    }
  }

  // submit input to API call
  handleSubmit = (e) => {
    e.preventDefault();
    this.props.getAPIData(this.state.zipcode);
  };

  render() {
    const error = this.validateZipcode();

    return (
      <section className="wrapperForm">
        <form
          className="zipcodeForm"
          onSubmit={(e) => {
            this.handleSubmit(e);
          }}
        >
          <label htmlFor="zipcode">Enter a zipcode</label>{" "}
          <input
            type={"number"}
            id={"zipcode"}
            name={"zipcode"}
            onChange={(e) => this.updateZipcode(e.target.value)}
            value={this.state.zipcode}
          ></input>
          {this.state.touched && <ValidationError message={error} />}
          <div>
            <button
              className="btn"
              type="submit"
              disabled={this.validateZipcode()}
            >
              Go
            </button>
          </div>
        </form>
      </section>
    );
  }
}

export default ZipcodeForm;
