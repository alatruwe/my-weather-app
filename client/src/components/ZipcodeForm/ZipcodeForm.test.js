import React from "react";
import ReactDOM from "react-dom";
import ZipcodeForm from "./ZipcodeForm";
import renderer from "react-test-renderer";

describe("WeatherCity", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ZipcodeForm />, div);
    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<ZipcodeForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
