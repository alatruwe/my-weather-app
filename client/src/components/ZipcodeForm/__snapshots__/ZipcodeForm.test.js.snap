// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCity renders the UI as expected 1`] = `
<section
  className="wrapperForm"
>
  <form
    className="zipcodeForm"
    onSubmit={[Function]}
  >
    <label
      htmlFor="zipcode"
    >
      Enter a zipcode
    </label>
     
    <input
      id="zipcode"
      name="zipcode"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Go
      </button>
    </div>
  </form>
</section>
`;
